src/Package.php

```diff
@@ -142,23 +135,36 @@ class Package
      * $package->build();
      * $package->statusIs(Package::INITIALIZED); // true
      * $package->boot();
-     * $package->statusIs(Package::BOOTED); // true
+     * $package->statusIs(Package::STATUS_DONE); // true
      * </code>
      */
     public const STATUS_IDLE = 2;
+    public const STATUS_INIT = 3;
     public const STATUS_INITIALIZED = 4;
-    public const STATUS_MODULES_ADDED = 5;
-    public const STATUS_BOOTING = self::STATUS_MODULES_ADDED;
+    public const STATUS_BOOTING = 5;
     public const STATUS_READY = 7;
-    public const STATUS_BOOTED = 8;
+    public const STATUS_DONE = 8;
     public const STATUS_FAILED = -8;
 
```

Comment:
A bit incosistent, but we have:

> `STATUS_INIT` and afterwards `STATUS_IINITIALIZED`

but we have:

> `STATUS_BOOTING` and afterwards `STATUS_READY`

Why do we not use `STATUS_BOOT` and `STATUS_BOOTED` instead (its set to "deprecated"? 👀

Reply:
I also don't see a reason for both `READY` and `DONE`, and I agree that mirroring `INITIALIZING`/`INITIALIZED` and `BOOTING`/`BOOTED` makes sense.

Reply:
The names of the statuses are inconsistent, yes, but they are consistent with the actions.

As agreed on #47 we have now a clea map netween statuses and actions fired.

**When `ACTION_INIT` is fired, the package is in `STATUS_INIT`**. Note that `ACTION_INIT` is existing, and it is probably the most used hook of the package. So when i had to have a status for it I wanted to use `STATUS_INIT`.

I had 3 options:

- Keep `ACTION_INIT` and introduce `STATUS_INITIALIZING`. Status names are more consistent, but no corrispondence action-status.
- Introduce `ACTION_INITIALIZING` and introduce `STATUS_INITIALIZING`. Status names are more consistent, action-status is also consistent, but requires to deprecate `ACTION_INIT` that is the most used hook.
- Keep `ACTION_INIT` and introduce `STATUS_INIT. Status names not consistent, but action-status is consistent.

I decided to use the third approach, because based on what was discussed in #47 having a consistent action-status mapping had priority.

The exact same thing for `STATUS_READY`.

**When `ACTION_READY` is fired, the package is in `STATUS_READY`**. Once again, the action constant was preexistent. The status name was `STATUS_BOOTED`. The 3 cases above apply again. And once again I decided to go with the 3rd option and introduce a new status name (deprecating teh previous), and so favor action-status corrispondence over the "internal consistency" of status names.

But I am fine also in having the first option, where we keep the action names (I don't want to deprecate those because it affect many packages), we keep consisten status names, but we don't have a corrispondence between status and actions.

Regarding the reason for both `READY` and `DONE`. Again, `READY` is the status the package has when external code can access its instance via the `READY` hook.
`DONE` means that no change of any kind can be done on the package and that it can not be accessed anymore via any hook.

Reply:
There's also a middle ground:


- `STATUS_IDLE => no action
- `STATUS_INITIALIZING` => `ACTION_INIT`
- `STATUS_INITIALIZED` => `STATUS_INITIALIZED`
- `STATUS_BOOTING` => no action
- `STATUS_BOOTED` => `ACTION_BOOTED`
- `STATUS_DONE` => no action

TSo we keep the status names consistent, we keep the status-to-name consistent 2 out of 3 times, and we deprecate `ACTION_READY` that is not very used. From a quick research, [it seems is only used once outside of this package](https://github.com/search?q=org%3Ainpsyde+OR+org%3AInpsyde-Global-Service-Provider++%22Package%3A%3AACTION_READY%22&type=code).

Reply:
Ok, because I saw both your thumbs up, I implemented the "middle ground" solution described above.